File: app/src/main/java/com/remotearthsolutions/expensetracker/fragments/OptionBottomSheetFragment.java
package com.remotearthsolutions.expensetracker.fragments;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import com.google.android.material.bottomsheet.BottomSheetDialogFragment;
import com.remotearthsolutions.expensetracker.R;

public class OptionBottomSheetFragment extends BottomSheetDialogFragment implements View.OnClickListener {

    private OptionBottomSheetFragment.Callback callback;
    private OptionBottomSheetFragment.OptionsFor optionsFor;

    public void setCallback(Callback callback, OptionBottomSheetFragment.OptionsFor optionsFor) {
        this.callback = callback;
        this.optionsFor = optionsFor;
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_optionbottomsheet, container, false);
        LinearLayout addAmountBtn = view.findViewById(R.id.addAmountBtn);
        addAmountBtn.setOnClickListener(this);
        view.findViewById(R.id.editBtn).setOnClickListener(this);
        view.findViewById(R.id.deleteBtn).setOnClickListener(this);

        switch (optionsFor){
            case CATEGORY:
                addAmountBtn.setVisibility(View.GONE);
                break;
            case ACCOUNT:
                addAmountBtn.setVisibility(View.VISIBLE);
                break;
        }

        return view;
    }

    @Override
    public void onClick(View v) {
        if (callback == null)
            return;

        switch (v.getId()) {
            case R.id.addAmountBtn:
                callback.onClickAddAmountBtn();
                break;
            case R.id.editBtn:
                callback.onClickEditBtn();
                break;
            case R.id.deleteBtn:
                callback.onClickDeleteBtn();
                break;
        }

        dismiss();
    }

    public interface Callback {
        void onClickAddAmountBtn();

        void onClickEditBtn();

        void onClickDeleteBtn();
    }

    public enum OptionsFor{
        CATEGORY,ACCOUNT
    }
}

File: app/src/main/java/com/remotearthsolutions/expensetracker/fragments/OptionBottomSheetFragment.kt
package com.remotearthsolutions.expensetracker.fragments

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.google.android.material.bottomsheet.BottomSheetDialogFragment
import com.remotearthsolutions.expensetracker.R
import com.remotearthsolutions.expensetracker.databinding.FragmentOptionbottomsheetBinding

class OptionBottomSheetFragment : BottomSheetDialogFragment(),
    View.OnClickListener {
    private lateinit var binding: FragmentOptionbottomsheetBinding
    private var callback: Callback? = null
    private var optionsFor: OptionsFor? = null
    fun setCallback(
        callback: Callback?,
        optionsFor: OptionsFor?
    ) {
        this.callback = callback
        this.optionsFor = optionsFor
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        binding = FragmentOptionbottomsheetBinding.inflate(layoutInflater, container, false)
        binding.addAmountBtn.setOnClickListener(this)
        binding.updateAmountBtn.setOnClickListener(this)
        binding.editBtn.setOnClickListener(this)
        binding.deleteBtn.setOnClickListener(this)
        when (optionsFor) {
            OptionsFor.CATEGORY -> {
                binding.addAmountBtn.visibility = View.GONE
                binding.underLine.visibility = View.GONE
                binding.updateAmountBtn.visibility = View.GONE
                binding.underLine1.visibility = View.GONE
            }
            OptionsFor.ACCOUNT -> {
                binding.addAmountBtn.visibility = View.VISIBLE
                binding.underLine.visibility = View.VISIBLE
                binding.updateAmountBtn.visibility = View.VISIBLE
                binding.underLine1.visibility = View.VISIBLE
            }
            null -> TODO()
        }
        return binding.root
    }

    override fun onClick(v: View) {
        if (callback == null) {
            return
        }
        when (v.id) {
            R.id.addAmountBtn -> callback!!.onClickAddAmountBtn()
            R.id.updateAmountBtn -> callback!!.onClickUpdateAmountBtn()
            R.id.editBtn -> callback!!.onClickEditBtn()
            R.id.deleteBtn -> callback!!.onClickDeleteBtn()
        }
        dismiss()
    }

    interface Callback {
        fun onClickAddAmountBtn() {}
        fun onClickUpdateAmountBtn() {}
        fun onClickEditBtn()
        fun onClickDeleteBtn()
    }

    enum class OptionsFor {
        CATEGORY, ACCOUNT
    }
}